src/develop/blend_gui.c

```diff
@@ -2940,7 +2940,8 @@ void dt_iop_gui_init_blending(GtkWidget *iopw, dt_iop_module_t *module)
     //box enclosing the mask mode selection buttons
     bd->masks_modes_box = GTK_BOX(gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0));
     //mask selection buttons packing in mask_box
-    for (int i = 0; i < g_list_length(bd->masks_modes_toggles); i++)
+    const int ntoggles = g_list_length(bd->masks_modes_toggles);
+    for (int i = 0; i < ntoggles; i++)
       gtk_box_pack_start(GTK_BOX(bd->masks_modes_box), GTK_WIDGET(g_list_nth_data(bd->masks_modes_toggles, i)), TRUE, TRUE, 0);
```

Comment:
for this one, I think the right way is to do something like : 
```
GList *l = bd->masks_modes_toggles;
while(l)
{
  gtk_box_pack_start(GTK_BOX(bd->masks_modes_box), GTK_WIDGET(l->data), TRUE, TRUE, 0);
  l = g_list_next(l);
}
```
as iirc `g_list_nth_data` scan the whole list each time...